core/frontend/src/AccuDraw.ts

```diff
@@ -951,43 +961,60 @@ export class AccuDraw {
           return BentleyStatus.ERROR;
       }
 
+    let parseResult;
+
     switch (index) {
       case ItemField.DIST_Item:
-        if (BentleyStatus.SUCCESS !== this.stringToUORs([this._distance], input))
-          return BentleyStatus.ERROR;
-        break;
+        parseResult = this.stringToDistance(input);
+        if (undefined !== parseResult.value && parseResult?.status === QuantityStatus.Success) {

```

Comment:
`stringToDistance` and `stringToAngle` return `ParseResult`, never `undefined` - why all the `undefined` checks?

Reply:
Fixed - I should have removed those '?' after I decided to ensure a ParseResult was returned.

Reply:
You are still unnecessarily comparing to `undefined`. Please don't resolve other people's comments.

Reply:
Sorry didn't know the protocol a clicking "Resolve".
Since Custom Parsers can be written, some parser could improperly return Success and undefined, so I was trying to avoid that situation. But I can remove that safeguard.

Reply:
Would be preferable to define `ParseResult` such that nonsensical combinations like that are not possible.

Reply:
Do you have any good examples to follow?

Reply:
[Something like this](https://www.typescriptlang.org/play?#code/KYOwrgtgBAimCGIAuBLJBPAyk+SwGcoBvKTMAY3OH3wBooBRAJyYHsmoBfAWACgMADsCgAFeE3zAAStTAAbJFAC8xKADd4csMABcUcBABGwDpygAfVfhx58euIlQZsuAlwDcfPgDMwIcqisIFAC4pIAFCggAmBIetZMUQDmAJR6YhLSsgrEfFBQTMB4TMEARKwA1qXKSipRMYoA-KoaWrpQAKxcUHok1q52sAjIaFg2BAB0zGymnrycQA)?
You still have the awkwardness of "Success" being an "error code".

Reply:
Or like keyin parse result: https://github.com/imodeljs/imodeljs/blob/80f159229b0b7154f8330c15f17540255d718038/core/frontend/src/tools/Tool.ts#L776

Reply:
I was thinking about solutions like that too...., but either would be a breaking change to the interface that is 2 years old... The interface is marked as alpha so I guess I could legally break it.

Reply:
@calebmshafer - since you approve changes in the quantity package, do you think I should break the ParseResult interface in the quantity package to address the issue Paul raises.

Reply:
I agree with @pmconne, it'd be a lot cleaner this way.  It's alpha so I'd say now is the time to break it as we'll probably want to start moving it to beta in the next couple months as the Units/formatting support moves forward.

Reply:
@pmconne and @calebmshafer - requested API changes ready for review.

Reply:
Thanks. Minor change needed in documentation comments.